Fix possible division by zero by clamping latitude to 85.05°

oxidase opened the pull request:
# Issue

Resolves #3530 by pre clamping of the input latitude.

## Tasklist
 - [x] review
 - [x] adjust for comments

TheMarex commented on include/util/web_mercator.hpp:
```diff
@@ -41,10 +41,11 @@ inline FloatLatitude yToLat(const double y)
-    const double y = detail::RAD_TO_DEGREE * 0.5 * std::log((1 + f) / (1 - f));
-    const auto clamped_y = std::max(-180., std::min(180., y));
-    return clamped_y;
+    const double epsg3857_limit = 85.051128779806592378; // 90(4 atan(exp(pi)) / pi - 1)
```
What about making this constant a `constexpr`? Like `static const constexpr double EPSG3857_MAX_LAT = 85.051128779806592378; // 90(4 atan(exp(pi)) / pi - 1);`

TheMarex commented on unit_tests/util/coordinate_calculation.cpp:
```diff
@@ -118,23 +118,20 @@ BOOST_AUTO_TEST_CASE(compute_angle)
-    BOOST_CHECK_THROW( coordinate_calculation::computeAngle(first,middle,end),
-                       boost::numeric::positive_overflow);
-                       */
+    BOOST_CHECK_THROW(
```
Good catch!

oxidase commented:
Rebased and changed `MAX_LATITUDE` to `EPSG3857_MAX_LATITUDE`. It influences also `xyzToMercator` where latitude also should be clamped by EPSG:3857 max latitude value and not 85

oxidase commented:
Update `web_mercator_tests/xyz_to_mercator` test expectations: previous values 19971868.8804085782 and and 19971868.880408578  mean 0 tile y-extent in Mercator projection. Correct maxy value is 6378137*pi or 20037508.342789277 in Mercator projection or 85.0511 in WGS
